code/datums/components/riding.dm

```diff
@@ -50,6 +54,32 @@
 	M.set_glide_size(movable_parent.glide_size)
 	M.updating_glide_size = FALSE
 	handle_vehicle_offsets(movable_parent.dir)
+	if(can_use_abilities)
+		setup_abilities(M)
+
+///Gives the rider the riding parent's abilities
+/datum/component/riding/proc/setup_abilities(mob/living/M)
+
+	if(!istype(parent, /mob/living))
+		return
+
+	var/mob/living/ridden_creature = parent
+
+	for(var/i in ridden_creature.abilities)
```

Comment:
Better variable names please.

Reply:
check the line below, this is normal behavior for these kinds of loops.

Reply:
Yep, the line bellow has an actually readable variable name.